android/engine/src/main/java/org/smartregister/fhircore/engine/util/SecureSharedPreference.kt

```diff
@@ -42,10 +43,22 @@ class SecureSharedPreference @Inject constructor(@ApplicationContext val context
   private fun getMasterKey() =
     MasterKey.Builder(context).setKeyScheme(MasterKey.KeyScheme.AES256_GCM).build()
 
-  fun saveCredentials(authCredentials: AuthCredentials) {
+  fun saveCredentials(username: String, password: CharArray) {
+
+    val randomSaltBytes = getRandomBytesOfSize(256)
+
     secureSharedPreferences.edit {
-      putString(SharedPreferenceKey.LOGIN_CREDENTIAL_KEY.name, authCredentials.encodeJson())
+      putString(
+        SharedPreferenceKey.LOGIN_CREDENTIAL_KEY.name,
+        AuthCredentials(
+            username,
+            Base64.getEncoder().encodeToString(randomSaltBytes),
+            password.toPasswordHash(randomSaltBytes)
+          )
```

Comment:
Add names to arguments (All the properties of `AuthCredentials` are of String type)

```suggestion
          AuthCredentials(
            username = username,
            salt = Base64.getEncoder().encodeToString(randomSaltBytes),
            passwordHash = password.toPasswordHash(randomSaltBytes)
          )
```